core/ledger/kvledger/upgrade_dbs.go

```diff
@@ -24,29 +32,39 @@ func UpgradeDBs(rootFSPath string) error {
 	defer fileLock.Unlock()
 
 	logger.Infof("Ledger data folder from config = [%s]", rootFSPath)
+	dbPath := LedgerProviderPath(rootFSPath)
+	db := leveldbhelper.CreateDB(&leveldbhelper.Conf{DBPath: dbPath})
+	db.Open()
+	defer db.Close()
 
-	// upgrades idStore (ledgerProvider) and drop databases
-	if err := UpgradeIDStoreFormat(rootFSPath); err != nil {
+	idStore := &idStore{db, dbPath}
+	format, err := idStore.getFormat()
+	if err != nil {
 		return err
 	}
+	if bytes.Equal(format, []byte(dataformat.Version20)) {
+		logger.Info("Ledger data format is current, do nothing.")
+		return nil
+	}
 
-	if err := dropDBs(rootFSPath); err != nil {
+	if err := dropCouchAndLevelDBs(config); err != nil {
 		return err
 	}
 
-	blockstorePath := BlockStorePath(rootFSPath)
-	return fsblkstorage.DeleteBlockStoreIndex(blockstorePath)
-}
+	if err := fsblkstorage.DeleteBlockStoreIndex(BlockStorePath(rootFSPath)); err != nil {
+		return err
+	}
 
-// UpgradeIDStoreFormat upgrades the format for idStore
-func UpgradeIDStoreFormat(rootFSPath string) error {
-	logger.Debugf("Attempting to upgrade idStore data format to current format %s", dataformat.Version20)
+	logger.Info("Updating the ledger provider database format ...")
+	return idStore.upgradeFormat()
+}
 
-	dbPath := LedgerProviderPath(rootFSPath)
-	db := leveldbhelper.CreateDB(&leveldbhelper.Conf{DBPath: dbPath})
-	db.Open()
-	defer db.Close()
+func dropCouchAndLevelDBs(config *ledger.Config) error {
```

Comment:
Prefer to use a separate PR for reset/rollback because they have user impact. As of dropDBs unnecessarily drops stateleveldb in the case of couchdb, it is true for other commands too. I can implement a new method instead of calling dropDBs.

Reply:
IMO, better to leave `dropDBs` as is. Instead, rename drop `dropStateLevelDB` as `dropStateDB`. Within `dropStateLevelDB`, you can check and delete one of the leveldb or couchdb.

Reply:
The suggestion given by Manish seems better.

Reply:
Yes, agree with Manish's suggestion, but it means that we will drop statecouchdb for all reset/rollback/upgrade commands. Meanwhile, I try to limit the scope of this PR. Maybe we can use this PR to update dropDBs (and dropStateDB) for all 3 commands and use a separate PR to update integration tests. What do you think?

Reply:
Yes, we can limit the scope of this PR as it simplifies the reviewer's task too.